app/web_framework/cluster_master_application.py

```diff
@@ -30,38 +30,71 @@ def __init__(self, cluster_master):
-                            RouteNode(r'subjob', _SubjobsHandler, 'subjobs').add_children([
-                                RouteNode(r'(\d+)', _SubjobHandler, 'subjob').add_children([
-                                    RouteNode(r'atom', _AtomsHandler, 'atoms').add_children([
-                                        RouteNode(r'(\d+)', _AtomHandler, 'atom').add_children([
-                                            RouteNode(r'console', _AtomConsoleHandler)
-                                        ])
-                                    ]),
-                                    RouteNode(r'result', _SubjobResultHandler)
-                                ])
+
+        api_v1 = [
+            RouteNode(r'v1', _APIVersionOneHandler).add_children([
+                RouteNode(r'metrics', _MetricsHandler),
+                RouteNode(r'version', _VersionHandler),
+                RouteNode(r'build', _BuildsHandler, 'builds').add_children([
+                    RouteNode(r'(\d+)', _BuildHandler, 'build').add_children([
+                        RouteNode(r'result', _BuildResultRedirectHandler),
+                        RouteNode(r'artifacts.tar.gz', _BuildTarResultHandler),
+                        RouteNode(r'artifacts.zip', _BuildZipResultHandler),
+                        RouteNode(r'subjob', _SubjobsHandler, 'subjobs').add_children([
+                            RouteNode(r'(\d+)', _SubjobHandler, 'subjob').add_children([
+                                RouteNode(r'atom', _AtomsHandler, 'atoms').add_children([
+                                    RouteNode(r'(\d+)', _AtomHandler, 'atom').add_children([
+                                        RouteNode(r'console', _AtomConsoleHandler)
+                                    ])
+                                ]),
+                                RouteNode(r'result', _SubjobResultHandler)
                             ])
                         ])
+                    ])
+                ]),
+                RouteNode(r'queue', _QueueHandler),
+                RouteNode(r'slave', _SlavesHandler, 'slaves').add_children([
+                    RouteNode(r'(\d+)', _SlaveHandler, 'slave').add_children([
+                        RouteNode(r'shutdown', _SlaveShutdownHandler, 'shutdown')
                     ]),
-                    RouteNode(r'queue', _QueueHandler),
-                    RouteNode(r'slave', _SlavesHandler, 'slaves').add_children([
-                        RouteNode(r'(\d+)', _SlaveHandler, 'slave').add_children([
-                            RouteNode(r'shutdown', _SlaveShutdownHandler, 'shutdown')
-                        ]),
-                        RouteNode(r'shutdown', _SlavesShutdownHandler, 'shutdown')
-                    ]),
-                    RouteNode(r'eventlog', _EventlogHandler)
+                    RouteNode(r'shutdown', _SlavesShutdownHandler, 'shutdown')
+                ]),
+                RouteNode(r'eventlog', _EventlogHandler)
+            ])]
+
+        api_v2 = [
```

Comment:
Like mentioned above, this is mainly for testing purposes right now to prove that it works. We'd probably want to remove this from this PR if we want to merge it at some point